runtime/src/accounts_db.rs

```diff
@@ -4560,26 +4577,34 @@ impl AccountsDb {
             } else {
                 (0, 0)
             };
-        let old_slots = self.accounts_cache.find_older_frozen_slots(MAX_CACHE_SLOTS);
-        let excess_slot_count = old_slots.len();
+
+        let mut excess_slot_count = 0;
         let mut unflushable_unrooted_slot_count = 0;
         let max_flushed_root = self.accounts_cache.fetch_max_flush_root();
-        let old_slot_flush_stats: Vec<_> = old_slots
-            .into_iter()
-            .filter_map(|old_slot| {
-                // Don't flush slots that are known to be unrooted
-                if old_slot > max_flushed_root {
-                    Some(self.flush_slot_cache(old_slot, None::<&mut fn(&_, &_) -> bool>))
-                } else {
-                    unflushable_unrooted_slot_count += 1;
-                    None
-                }
-            })
-            .collect();
-        info!(
-            "req_flush_root: {:?} old_slot_flushes: {:?}",
-            requested_flush_root, old_slot_flush_stats
-        );
+        if self.should_aggressively_flush_cache() {
+            let old_slots = self.accounts_cache.find_older_frozen_slots();
+            excess_slot_count = old_slots.len();
+            let old_slot_flush_stats: Vec<_> = old_slots
+                .into_iter()
+                .filter_map(|old_slot| {
+                    // Don't flush slots that are known to be unrooted
+                    if old_slot > max_flushed_root {
+                        if self.should_aggressively_flush_cache() {
+                            Some(self.flush_slot_cache(old_slot, None::<&mut fn(&_, &_) -> bool>))
+                        } else {
+                            None
+                        }
+                    } else {
+                        unflushable_unrooted_slot_count += 1;
+                        None
+                    }
+                })
+                .collect();
+            info!(
+                "req_flush_root: {:?} old_slot_flushes: {:?}",
+                requested_flush_root, old_slot_flush_stats
+            );
```

Comment:
Might make sense to turn this into a datapoint capturing:

1. Number of slots in the cache at the time the aggressive flush happened
2. An aggregate of the measures in `old_slot_flush_stats`

Reply:
aggregated.